### ISSUE TITLE
`Equalization` channel dimension is not calculated during trace time

### ISSUE THREAD

**Issue by quantumalaviya** [keras-team/keras-cv#431]
There are a few issues I faced while trying to use the `Equalization` layer with AugMix in #407 

The channel dimension is not calculated during trace time (I believe due to the `vectorized_map` from #201). This makes it incompatible with other layers in a chain. Moreover, this is the only layer that gets traced every time it's called. It's not ideal and repeated occurrences of this layer are very inefficient.

**Comment by LukeWood** [comment]
Definitely, Equalization should be updated.  Thanks for doing some digging here.

I believe these changes were introduced by my PR that adds support for dynamic channel sizes; but this behavior is very undesirable.
